lib/MetaCPAN/Script/Release.pm

```diff
@@ -202,36 +202,25 @@ sub run {
 
 sub import_tarball {
     my ( $self, $tarball ) = @_;
+
+    $tarball = MetaCPAN::Model::Tarball->new( tarball => $tarball );
     my $cpan = $self->index;
 
-    $tarball = Path::Class::File->new($tarball);
     my $d = CPAN::DistnameInfo->new($tarball);
     my ( $author, $archive, $name )
         = ( $d->cpanid, $d->filename, $d->distvname );
-    log_info {"Processing $tarball"};
-
-    # load Archive::Any in the child due to bugs in MMagic and MIME::Types
-    require Archive::Any;
-    my $at = Archive::Any->new($tarball);
-    my $tmpdir
-        = dir( File::Temp::tempdir( CLEANUP => 0, DIR => $self->base_dir ) );
-
-    log_error {"$tarball is being impolite"} if $at->is_impolite;
+    $tarball->author = $author;
```

Comment:
This actually won't work either since the attribute is `ro`.  If you the object needs this attribute, you should pass it when the object is created, via `new( author => 'foo', ... )`